mx.sulong/mx_sulong.py

```diff
@@ -1132,18 +1132,22 @@ def dos2unix(args=None):
 
 def genInlineAssemblyParser(args=None, out=None):
     """generate inline assembly parser and scanner if corresponding grammer is new"""
-    generatedParserDir = _inlineAssemblySrcDir + _inlineAssemblyPackageName.replace('.', '/')
-    generatedParserFile = generatedParserDir + "/Parser.java"
-    generatedScannerFile = generatedParserDir + "/Scanner.java"
-    if not isfile(generatedParserFile) or not isfile(generatedScannerFile) or os.path.getmtime(_inlineAssemblyGrammer) >= os.path.getmtime(generatedParserFile):
+    generatedParserDir = _inlineAssemblySrcDir + _inlineAssemblyPackageName.replace('.', '/') + "/"
+    generatedFiles = [generatedParserDir + "Parser.java", generatedParserDir + "Scanner.java"]
+    configFiles = [_inlineAssemblySrcDir + "InlineAssembly.atg", _inlineAssemblySrcDir + "Parser.frame", _inlineAssemblySrcDir + "Scanner.frame", _inlineAssemblySrcDir + "Copyright.frame"]
+    isAllGeneratedFilesExists = all([isfile(fileName) for fileName in generatedFiles])
+    latestGeneratedFile = sorted(generatedFiles, key=os.path.getmtime, reverse=True)[0] if isAllGeneratedFilesExists else ""
+    latestConfigFile = sorted(configFiles, key=os.path.getmtime, reverse=True)[0]
+    #If any auto-generated file is missing or any config file is updated after last auto-generation then regenerate the files
+    if not isAllGeneratedFilesExists or os.path.getmtime(latestConfigFile) >= os.path.getmtime(latestGeneratedFile):
```

Comment:
It is great to have the parser generated only if you detect changes in the ATG or frames!